surveys/surveys.go

```diff
@@ -68,28 +87,39 @@ func LauncherSchemaFromFilename(filename string) LauncherSchema {
 
 // GetAvailableSchemas Gets the list of static schemas an joins them with any schemas from the eq-survey-register if defined
 func GetAvailableSchemas() LauncherSchemas {
+
+	client := clients.GetHTTPClient()
+
 	schemaList := LauncherSchemas{}
 
-	runnerSchemas := getAvailableSchemasFromRunner()
-
-	for _, launcherSchema := range runnerSchemas {
-		if strings.HasPrefix(launcherSchema.Name, "test_") {
-			schemaList.Test = append(schemaList.Test, launcherSchema)
-		} else if strings.HasPrefix(launcherSchema.Name, "census_") {
-			schemaList.Census = append(schemaList.Census, launcherSchema)
-		} else if strings.HasPrefix(launcherSchema.Name, "lms_") {
-			schemaList.Social = append(schemaList.Social, launcherSchema)
-		} else {
-			schemaList.Business = append(schemaList.Business, launcherSchema)
+	runnerSchemas, err := getAvailableSchemasFromRunner(client)
+	if err != nil {
+		log.Printf(`WARN: Unexpected error whilst retrieving schemas from EQRunner; skipping repository`)
+	} else {
```

Comment:
Im not sure if my interpretation of the code here is correct, but could we stop executing the code if error on line 96 is true? I imagine it will fail if it keeps going anyways?

Reply:
We can't stop executing the code as it needs to try getting schemas from Runner and the Register. If there is an error, it'll stop getting the schemas from that repository and move on to the next. e.g, stop getting them from runner and try the Register instead.

Reply:
Okay, that makes sense.